### PR TITLE
refactor ExtraHeaderSigVerifierHolder

### PR DESCRIPTION
**Pull request by axenteoctavian**
## Reasoning behind the pull request
- refactor `ExtraHeaderSigVerifierHolder`
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: cmd/sovereignnode/sovereignNodeRunner.go [hunk 1]
```diff
@@ -1301,49 +1301,37 @@ func (snr *sovereignNodeRunner) CreateManagedProcessComponents(
...
-	extraHeaderSigVerifierHolder := headerCheck.NewExtraHeaderSigVerifierHolder()
-	sovHeaderSigVerifier, err := headerCheck.NewSovereignHeaderSigVerifier(cryptoComponents.BlockSigner())
-	if err != nil {
-		return nil, err
-	}
-
-	err = extraHeaderSigVerifierHolder.RegisterExtraHeaderSigVerifier(sovHeaderSigVerifier)
-	if err != nil {
-		return nil, err
-	}
```
Comment (mariusmihaic) [inline cmd/sovereignnode/sovereignNodeRunner.go:1313]: This whole code should've happen inside `sovereignRunTypeComp` -> `Create`

File: cmd/sovereignnode/sovereignNodeRunner.go [hunk 2]
```diff
@@ -1664,11 +1652,17 @@ func (snr *sovereignNodeRunner) CreateArgsRunTypeComponents(coreComp mainFactory
 		return nil, fmt.Errorf("NewRunTypeComponentsFactory failed: %w", err)
 	}
 
+	sovHeaderSigVerifier, err := headerCheck.NewSovereignHeaderSigVerifier(cryptoComp.BlockSigner())
+	if err != nil {
+		return nil, err
+	}
```
Comment (mariusmihaic) [inline cmd/sovereignnode/sovereignNodeRunner.go:1658]: This is **NOT OK**.

Why would you create it outside runtype and then pass it on args? you need to create it inside `sovRunTypeComp`.
For this you probably need to pass crypto components to run type sov

File: factory/runType/sovereignRunTypeComponents.go [hunk 3]
```diff
@@ -37,13 +38,15 @@ type ArgsSovereignRunTypeComponents struct {
...
 	TopicsChecker            sovereign.TopicsCheckerHandler
+	ExtraVerifier            process.ExtraHeaderSigVerifierHandler
 }
 
 type sovereignRunTypeComponentsFactory struct {
 	*runTypeComponentsFactory
 	cfg           config.SovereignConfig
 	dataCodec     sovereign.DataCodecHandler
 	topicsChecker sovereign.TopicsCheckerHandler
+	extraVerifier process.ExtraHeaderSigVerifierHandler
```
Comment (mariusmihaic) [inline factory/runType/sovereignRunTypeComponents.go:49]: This should be deleted

**Review by mariusmihaic — CHANGES_REQUESTED** [review]

File: cmd/sovereignnode/sovereignNodeRunner.go (hunk 1 above)
Comment (axenteoctavian) [reply cmd/sovereignnode/sovereignNodeRunner.go:1313]: To minimize the conflicts with the next PRs, I created another PR for this at the end of refactor 
https://github.com/multiversx/mx-chain-go/pull/6140

File: cmd/sovereignnode/sovereignNodeRunner.go (hunk 2 above)
Comment (axenteoctavian) [reply cmd/sovereignnode/sovereignNodeRunner.go:1658]: To minimize the conflicts with the next PRs, I created another PR for this at the end of refactor 
https://github.com/multiversx/mx-chain-go/pull/6140

File: factory/runType/sovereignRunTypeComponents.go (hunk 3 above)
Comment (axenteoctavian) [reply factory/runType/sovereignRunTypeComponents.go:49]: To minimize the conflicts with the next PRs, I created another PR for this at the end of refactor 
https://github.com/multiversx/mx-chain-go/pull/6140

**Review by mariusmihaic — DISMISSED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**